scalafmt-tests/src/test/scala-2.12/org/scalafmt/ScalafmtProps.scala

```diff
@@ -33,7 +34,14 @@ class ScalafmtProps extends FormatAssertions {
           case Formatted.Success(formatted) =>
             assertFormatPreservesAst[Source](code, formatted)
             val formattedSecondTime = Scalafmt.format(formatted, config).get
-            assertNoDiff(formattedSecondTime, formatted, "Idempotency")
+            Diffs.assertNoDiff(formattedSecondTime, formatted, diff => {
```

Comment:
I would ideally like to avoid using internal APIs, what is missing from the public api?

Reply:
We need diff report [here](https://github.com/scalameta/scalafmt/pull/1681/files/0231ae14725abf8f68e36ecc3d8abcbbcd688e1a#diff-44d568678882dc62a2d6ff89d804a8eeR77). I will try to extract it from `FailException`'s [message](https://github.com/scalameta/munit/blob/master/munit/shared/src/main/scala/munit/Assertions.scala#L55)

Reply:
done